schemas/iso19139/src/main/plugin/iso19139/index-fields/default.xsl

```diff
@@ -330,53 +336,62 @@
 
         <xsl:for-each select="$listOfKeywords">
           <xsl:variable name="keyword" select="string(.)"/>
-
+          <xsl:variable name="keyURI" select="./@xlink:href"/>
           <Field name="keyword" string="{$keyword}" store="true" index="true"/>
 
           <!-- If INSPIRE is enabled, check if the keyword is one of the 34 themes
                and index annex, theme and theme in english. -->
-          <xsl:if test="$inspire='true' and normalize-space(lower-case($thesaurusName)) = 'gemet - inspire themes, version 1.0'">
-
-            <xsl:if test="string-length(.) &gt; 0">
-
-              <xsl:variable name="inspireannex">
-                <xsl:call-template name="determineInspireAnnex">
-                  <xsl:with-param name="keyword" select="$keyword"/>
-                  <xsl:with-param name="inspireThemes" select="$inspire-theme"/>
-                </xsl:call-template>
-              </xsl:variable>
-
-              <xsl:variable name="inspireThemeAcronym">
-                <xsl:call-template name="getInspireThemeAcronym">
-                  <xsl:with-param name="keyword" select="$keyword"/>
-                </xsl:call-template>
-              </xsl:variable>
-
-              <!-- Add the inspire field if it's one of the 34 themes -->
-              <xsl:if test="normalize-space($inspireannex)!=''">
-                <Field name="inspiretheme" string="{$keyword}" store="true" index="true"/>
-                <Field name="inspirethemewithac"
-                       string="{concat($inspireThemeAcronym, '|', $keyword)}"
-                       store="true" index="true"/>
-
-                <!--<Field name="inspirethemeacronym" string="{$inspireThemeAcronym}" store="true" index="true"/>-->
-                <xsl:variable name="inspireThemeURI"
-                              select="$inspire-theme[skos:prefLabel = $keyword]/@rdf:about"/>
-                <Field name="inspirethemeuri" string="{$inspireThemeURI}" store="true"
-                       index="true"/>
-
-                <xsl:variable name="englishInspireTheme">
-                  <xsl:call-template name="translateInspireThemeToEnglish">
+          <xsl:if test="$inspire='true'">
+            <xsl:if test="$keyURI">
+              <xsl:for-each select="tokenize($inspire-thesauri-as-filter,',')">
```

Comment:
We already have a field for each thesaurus. See https://github.com/geonetwork/core-geonetwork/blob/master/schemas/iso19139/src/main/plugin/iso19139/index-fields/default.xsl#L403-L415

So you should be able to create a facet on spatial scope by simply adding 
something like:
```	  
<facet name="C19" label="C19" indexKey="thesaurus-external.reference-geographical-area.NVS.C19"/>
<facet name="atlantos_element" label="atlantos_element" indexKey="thesaurus-local.theme.atlantos_element"/>
```

Field is created with this rule `thesaurus-{{thesaurusIdentifier}}={{keyword}} `

Reply:
With Elasticsearch it will be much easier to notice that fields are available in the index mapping
![image](https://user-images.githubusercontent.com/1701393/80694583-fd901700-8ad4-11ea-909a-22bf9e716959.png)

and configure facets easily.

An issue with dynamic fields is that we may reach `index.mapping.total_fields.limit` limits at some points and this will require some improvements in 4.x ...